### ISSUE TITLE
Stopping the Robot

### ISSUE THREAD

**Issue by Tim2309** [nicoh88/homebridge-xiaomi-roborock-vacuum#105]
The bug "Stop #73" still exist with the S6
if you set the speed to 0% the robot stop and say "spot cleaning"
if you first pause the robot and then set to speed 0% to robot is going back to the dock

> Hi,
when I stop the vacuum with the HomeKit switch the robot stops and don't go to the dock.
is there a solution that due robot go to the dock?<

Here is a quick video:
https://youtu.be/SUCCrSIEvv0

**Comment by afharo** [comment]
Interesting... Thanks for the video!

I'd love to see the debug logs. If you could run the process with the environment variable `DEBUG=thing:miio:*,miio:*`, I think it will help us identify what commands are sent to the robot. 
The command would look something like 
```shell
DEBUG=thing:miio:*,miio:* homebridge -D
```

According to the current implementation, the logs should look something like:
```
ACT setCleaning | roborock.vacuum.s6 | Stop cleaning and go to charge.
thing:miio:96818260 -> (5) {"method":"app_stop","params":[],"id":89}
thing:miio:96818260 <- Message: `{"result":["ok"],"id":89}`
thing:miio:96818260 -> (5) {"method":"app_charge","params":[],"id":90}
thing:miio:96818260 <- Message: `{"result":["ok"],"id":90}`
```

**Comment by Tim2309** [comment]
I use Homebridge with the plugin/webinterface "Homebridge Config UI X" I activated the debug mode of the plugin

if I try it with your command in the terminal (in the webinterface) I get the following output:
/homebridge # DEBUG=thing:miio:*,miio:* homebridge -D
[2/6/2020, 5:00:22 PM] config.json (/root/.homebridge/config.json) not found.
[2/6/2020, 5:00:23 PM] Loaded plugin: homebridge-config-ui-x
[2/6/2020, 5:00:23 PM] Registering platform 'homebridge-config-ui-x.config'
[2/6/2020, 5:00:23 PM] ---

Heres the log created with the debug mode of the UI:

Turn it on then 0%:

[2/6/2020, 4:53:41 PM] [Staubsauger] DEB ensureDevice | roborock.vacuum.s6 | The socket is still on. Reusing it.
[2/6/2020, 4:53:41 PM] [Staubsauger] DEB ensureDevice | roborock.vacuum.s6 | The socket is still on. Reusing it.
[2/6/2020, 4:53:41 PM] [Staubsauger] DEB ensureDevice | roborock.vacuum.s6 | The socket is still on. Reusing it.
[2/6/2020, 4:53:41 PM] [Staubsauger] INF getPauseState | roborock.vacuum.s6 | Pause possible is false
[2/6/2020, 4:53:41 PM] [Staubsauger] INF getBatteryLow | roborock.vacuum.s6 | Batterylevel is 100%
[2/6/2020, 4:53:41 PM] [Staubsauger] INF getCleaning | roborock.vacuum.s6 | Cleaning is false
[2/6/2020, 4:53:41 PM] [Staubsauger] DEB ensureDevice | roborock.vacuum.s6 | The socket is still on. Reusing it.
[2/6/2020, 4:53:41 PM] [Staubsauger] DEB ensureDevice | roborock.vacuum.s6 | The socket is still on. Reusing it.
[2/6/2020, 4:53:41 PM] [Staubsauger] INF getSpeed | roborock.vacuum.s6 | Fanspeed is 101 over miIO. Converting to HomeKit
[2/6/2020, 4:53:41 PM] [Staubsauger] INF getSpeed | roborock.vacuum.s6 | Fanspeed is 101 over miIO "Quiet" > HomeKit speed 38%
[2/6/2020, 4:53:41 PM] [Staubsauger] INF getDocked | roborock.vacuum.s6 | Robot Docked is true (Status is charging)
[2/6/2020, 4:53:50 PM] [Staubsauger] DEB ensureDevice | roborock.vacuum.s6 | The socket is still on. Reusing it.
[2/6/2020, 4:53:50 PM] [Staubsauger] DEB ensureDevice | roborock.vacuum.s6 | The socket is still on. Reusing it.
[2/6/2020, 4:53:50 PM] [Staubsauger] ACT setCleaning | roborock.vacuum.s6 | Start cleaning, not charging.
[2/6/2020, 4:53:50 PM] [Staubsauger] ACT setSpeed | roborock.vacuum.s6 | Speed got 47% over HomeKit > CLEANUP.
[2/6/2020, 4:53:50 PM] [Staubsauger] ACT setSpeed | roborock.vacuum.s6 | FanSpeed set to 102 over miIO for "Balanced".
[2/6/2020, 4:53:50 PM] [Staubsauger] MON changedSpeed | roborock.vacuum.s6 | FanSpeed is now 102%
[2/6/2020, 4:53:50 PM] [Staubsauger] INF changedSpeed | roborock.vacuum.s6 | Speed was changed to 102% (Balanced), for HomeKit 60%
[2/6/2020, 4:53:51 PM] [Staubsauger] MON changedPause | roborock.vacuum.s6 | CleaningState is now undefined
[2/6/2020, 4:53:51 PM] [Staubsauger] INF changedPause | roborock.vacuum.s6 | Paused not possible, no cleaning
[2/6/2020, 4:53:51 PM] [Staubsauger] MON changedCharging | roborock.vacuum.s6 | ChargingState is now false
[2/6/2020, 4:53:51 PM] [Staubsauger] INF changedCharging | roborock.vacuum.s6 | Charging is cancelled
[2/6/2020, 4:53:51 PM] [Staubsauger] INF changedCharging | roborock.vacuum.s6 | Robot not anymore in dock.
[2/6/2020, 4:53:51 PM] [Staubsauger] MON changedCleaning | roborock.vacuum.s6 | CleaningState is now true
[2/6/2020, 4:53:51 PM] [Staubsauger] INF changedCleaning | roborock.vacuum.s6 | Cleaning is ON.
[2/6/2020, 4:53:51 PM] [Staubsauger] MON changedPause | roborock.vacuum.s6 | CleaningState is now true
[2/6/2020, 4:53:51 PM] [Staubsauger] INF changedPause | roborock.vacuum.s6 | Paused possible
[2/6/2020, 4:53:51 PM] [Staubsauger] DEB ensureDevice | roborock.vacuum.s6 | The socket is still on. Reusing it.
[2/6/2020, 4:53:51 PM] [Staubsauger] ACT setSpeed | roborock.vacuum.s6 | Speed got 58% over HomeKit > CLEANUP.
[2/6/2020, 4:53:51 PM] [Staubsauger] ACT setSpeed | roborock.vacuum.s6 | FanSpeed set to 102 over miIO for "Balanced".
[2/6/2020, 4:53:53 PM] [Staubsauger] DEB ensureDevice | roborock.vacuum.s6 | The socket is still on. Reusing it.
[2/6/2020, 4:53:53 PM] [Staubsauger] DEB getState | roborock.vacuum.s6 | State {"batteryLevel":100,"charging":false,"cleaning":true,"fanSpeed":102}
[2/6/2020, 4:53:53 PM] [Staubsauger] DEB changedBattery | roborock.vacuum.s6 | BatteryLevel 100%
[2/6/2020, 4:54:09 PM] [Staubsauger] DEB ensureDevice | roborock.vacuum.s6 | The socket is still on. Reusing it.
[2/6/2020, 4:54:09 PM] [Staubsauger] DEB ensureDevice | roborock.vacuum.s6 | The socket is still on. Reusing it.
[2/6/2020, 4:54:09 PM] [Staubsauger] ACT setCleaning | roborock.vacuum.s6 | Stop cleaning and go to charge.
[2/6/2020, 4:54:09 PM] [Staubsauger] ACT setSpeed | roborock.vacuum.s6 | Speed got 0% over HomeKit > CLEANUP.
[2/6/2020, 4:54:09 PM] [Staubsauger] ACT setSpeed | roborock.vacuum.s6 | FanSpeed set to 0 over miIO for "Off".
[2/6/2020, 4:54:10 PM] [Staubsauger] MON changedPause | roborock.vacuum.s6 | CleaningState is now undefined
[2/6/2020, 4:54:10 PM] [Staubsauger] INF changedPause | roborock.vacuum.s6 | Paused not possible, no cleaning
[2/6/2020, 4:54:10 PM] [Staubsauger] MON changedCleaning | roborock.vacuum.s6 | CleaningState is now false
[2/6/2020, 4:54:10 PM] [Staubsauger] INF changedCleaning | roborock.vacuum.s6 | Cleaning is OFF.
[2/6/2020, 4:54:10 PM] [Staubsauger] MON changedPause | roborock.vacuum.s6 | CleaningState is now false
[2/6/2020, 4:54:10 PM] [Staubsauger] INF changedPause | roborock.vacuum.s6 | Paused not possible, no cleaning


then wait for 1min start again, then pause and 0%:

[2/6/2020, 4:54:23 PM] [Staubsauger] DEB ensureDevice | roborock.vacuum.s6 | The socket is still on. Reusing it.
[2/6/2020, 4:54:23 PM] [Staubsauger] DEB getState | roborock.vacuum.s6 | State {"batteryLevel":100,"charging":false,"cleaning":false,"fanSpeed":102}
[2/6/2020, 4:54:23 PM] [Staubsauger] DEB changedBattery | roborock.vacuum.s6 | BatteryLevel 100%
[2/6/2020, 4:54:53 PM] [Staubsauger] DEB ensureDevice | roborock.vacuum.s6 | The socket is still on. Reusing it.
[2/6/2020, 4:54:53 PM] [Staubsauger] DEB getState | roborock.vacuum.s6 | State {"batteryLevel":100,"charging":false,"cleaning":false,"fanSpeed":102}
[2/6/2020, 4:54:53 PM] [Staubsauger] DEB changedBattery | roborock.vacuum.s6 | BatteryLevel 100%
[2/6/2020, 4:55:23 PM] [Staubsauger] DEB ensureDevice | roborock.vacuum.s6 | The socket is still on. Reusing it.
[2/6/2020, 4:55:23 PM] [Staubsauger] DEB getState | roborock.vacuum.s6 | State {"batteryLevel":100,"charging":false,"cleaning":false,"fanSpeed":102}
[2/6/2020, 4:55:23 PM] [Staubsauger] DEB changedBattery | roborock.vacuum.s6 | BatteryLevel 100%
[2/6/2020, 4:55:25 PM] [Staubsauger] DEB ensureDevice | roborock.vacuum.s6 | The socket is still on. Reusing it.
[2/6/2020, 4:55:25 PM] [Staubsauger] DEB ensureDevice | roborock.vacuum.s6 | The socket is still on. Reusing it.
[2/6/2020, 4:55:25 PM] [Staubsauger] DEB ensureDevice | roborock.vacuum.s6 | The socket is still on. Reusing it.
[2/6/2020, 4:55:25 PM] [Staubsauger] DEB ensureDevice | roborock.vacuum.s6 | The socket is still on. Reusing it.
[2/6/2020, 4:55:25 PM] [Staubsauger] DEB ensureDevice | roborock.vacuum.s6 | The socket is still on. Reusing it.
[2/6/2020, 4:55:25 PM] [Staubsauger] INF getSpeed | roborock.vacuum.s6 | Fanspeed is 102 over miIO. Converting to HomeKit
[2/6/2020, 4:55:25 PM] [Staubsauger] INF getSpeed | roborock.vacuum.s6 | Fanspeed is 102 over miIO "Balanced" > HomeKit speed 60%
[2/6/2020, 4:55:25 PM] [Staubsauger] INF getCleaning | roborock.vacuum.s6 | Cleaning is false
[2/6/2020, 4:55:25 PM] [Staubsauger] INF getBatteryLow | roborock.vacuum.s6 | Batterylevel is 100%
[2/6/2020, 4:55:25 PM] [Staubsauger] INF getPauseState | roborock.vacuum.s6 | Pause possible is false
[2/6/2020, 4:55:25 PM] [Staubsauger] INF getDocked | roborock.vacuum.s6 | Robot Docked is false (Status is waiting)
[2/6/2020, 4:55:26 PM] [Staubsauger] DEB ensureDevice | roborock.vacuum.s6 | The socket is still on. Reusing it.
[2/6/2020, 4:55:26 PM] [Staubsauger] DEB ensureDevice | roborock.vacuum.s6 | The socket is still on. Reusing it.
[2/6/2020, 4:55:26 PM] [Staubsauger] ACT setCleaning | roborock.vacuum.s6 | Start cleaning, not charging.
[2/6/2020, 4:55:26 PM] [Staubsauger] ACT setSpeed | roborock.vacuum.s6 | Speed got 37% over HomeKit > CLEANUP.
[2/6/2020, 4:55:26 PM] [Staubsauger] ACT setSpeed | roborock.vacuum.s6 | FanSpeed set to 101 over miIO for "Quiet".
[2/6/2020, 4:55:26 PM] [Staubsauger] MON changedSpeed | roborock.vacuum.s6 | FanSpeed is now 101%
[2/6/2020, 4:55:26 PM] [Staubsauger] INF changedSpeed | roborock.vacuum.s6 | Speed was changed to 101% (Quiet), for HomeKit 38%
[2/6/2020, 4:55:27 PM] [Staubsauger] MON changedPause | roborock.vacuum.s6 | CleaningState is now undefined
[2/6/2020, 4:55:27 PM] [Staubsauger] INF changedPause | roborock.vacuum.s6 | Paused not possible, no cleaning
[2/6/2020, 4:55:27 PM] [Staubsauger] MON changedCleaning | roborock.vacuum.s6 | CleaningState is now true
[2/6/2020, 4:55:27 PM] [Staubsauger] INF changedCleaning | roborock.vacuum.s6 | Cleaning is ON.
[2/6/2020, 4:55:27 PM] [Staubsauger] MON changedPause | roborock.vacuum.s6 | CleaningState is now true
[2/6/2020, 4:55:27 PM] [Staubsauger] INF changedPause | roborock.vacuum.s6 | Paused possible
[2/6/2020, 4:55:27 PM] [Staubsauger] DEB ensureDevice | roborock.vacuum.s6 | The socket is still on. Reusing it.
[2/6/2020, 4:55:27 PM] [Staubsauger] ACT setSpeed | roborock.vacuum.s6 | Speed got 53% over HomeKit > CLEANUP.
[2/6/2020, 4:55:27 PM] [Staubsauger] ACT setSpeed | roborock.vacuum.s6 | FanSpeed set to 102 over miIO for "Balanced".
[2/6/2020, 4:55:27 PM] [Staubsauger] MON changedSpeed | roborock.vacuum.s6 | FanSpeed is now 102%
[2/6/2020, 4:55:27 PM] [Staubsauger] INF changedSpeed | roborock.vacuum.s6 | Speed was changed to 102% (Balanced), for HomeKit 60%
[2/6/2020, 4:55:37 PM] [Staubsauger] DEB ensureDevice | roborock.vacuum.s6 | The socket is still on. Reusing it.
[2/6/2020, 4:55:38 PM] [Staubsauger] DEB ensureDevice | roborock.vacuum.s6 | The socket is still on. Reusing it.
[2/6/2020, 4:55:38 PM] [Staubsauger] DEB ensureDevice | roborock.vacuum.s6 | The socket is still on. Reusing it.
[2/6/2020, 4:55:38 PM] [Staubsauger] ACT setCleaning | roborock.vacuum.s6 | Stop cleaning and go to charge.
[2/6/2020, 4:55:38 PM] [Staubsauger] ACT setSpeed | roborock.vacuum.s6 | Speed got 0% over HomeKit > CLEANUP.
[2/6/2020, 4:55:38 PM] [Staubsauger] ACT setSpeed | roborock.vacuum.s6 | FanSpeed set to 0 over miIO for "Off".
[2/6/2020, 4:55:39 PM] [Staubsauger] MON changedPause | roborock.vacuum.s6 | CleaningState is now undefined
[2/6/2020, 4:55:39 PM] [Staubsauger] INF changedPause | roborock.vacuum.s6 | Paused not possible, no cleaning
[2/6/2020, 4:55:39 PM] [Staubsauger] MON changedCleaning | roborock.vacuum.s6 | CleaningState is now false
[2/6/2020, 4:55:39 PM] [Staubsauger] INF changedCleaning | roborock.vacuum.s6 | Cleaning is OFF.
[2/6/2020, 4:55:39 PM] [Staubsauger] MON changedPause | roborock.vacuum.s6 | CleaningState is now false
[2/6/2020, 4:55:39 PM] [Staubsauger] INF changedPause | roborock.vacuum.s6 | Paused not possible, no cleaning

**Comment by Tim2309** [comment]
I run homebridge on a Synology with docker, I don't know how to activate the debug mode you mentioned

**Comment by afharo** [comment]
Hmm, I think I might have a possible solution to that. Just pushed the update to git in the linked PR #106

**Comment by nicoh88** [comment]
Successfully published homebridge-xiaomi-roborock-vacuum@0.6.8

```
sudo su -
npm -g install homebridge-xiaomi-roborock-vacuum@latest
```

**Comment by Tim2309** [comment]
The problem still exist with 0.6.8
same behavior like 0.6.7
